staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -109,24 +114,11 @@ type Owner struct {
 // the controller is inactive. This can happen when kubelet is run stand-alone
 // without an apiserver. In that case we can't and don't need to publish
 // ResourceSlices.
-func StartController(ctx context.Context, kubeClient kubernetes.Interface, driver string, owner Owner, resources *DriverResources) *Controller {
-	if kubeClient == nil {
-		return nil
-	}
-
+func StartController(ctx context.Context, kubeClient kubernetes.Interface, driver string, owner Owner, resources *DriverResources) (*Controller, error) {
 	logger := klog.FromContext(ctx)
-	ctx, cancel := context.WithCancelCause(ctx)
-
-	c := &Controller{
-		cancel:     cancel,
-		kubeClient: kubeClient,
-		driver:     driver,
-		owner:      owner,
-		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
-			workqueue.DefaultTypedControllerRateLimiter[string](),
-			workqueue.TypedRateLimitingQueueConfig[string]{Name: "node_resource_slices"},
-		),
-		resources: resources,
+	c, err := newController(ctx, kubeClient, driver, owner, resources)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create controller: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("create controller: %w", err)
```

The only "failed to" (if any at all!) should be in the code which generates the initial error.

Reply:
done